app/src/main/java/com/tomaszpolanski/androidsandbox/ViewModelBase.kt

```diff
@@ -0,0 +1,25 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.arch.lifecycle.ViewModel
+import android.os.Parcelable
+import io.reactivex.Flowable
+import io.reactivex.Single
+import io.reactivex.processors.PublishProcessor
+
+abstract class ViewModelBase<S : Parcelable, in E>(
+    private val schedulerProvider: SchedulerProvider
+                                                  ) : ViewModel() {
+
+    private val intentPublishSubject = PublishProcessor.create<Pair<E, S?>>()
```

Comment:
why use a processor and call it subject? its a bit confusing. i don't think we need to handle backpressure here so its probably best to use a subject. :)

Reply:
If we have backpressure coming from UI events it likely indicates a bug. I've never seen a use case for backpressure in an android app. Backpressure is more for server side events and work queues.